Mathlib/RingTheory/KrullDimension.lean

```diff
@@ -0,0 +1,120 @@
+-/
+
+/--
+The ring theoretic Krull dimension is the Krull dimension of its spectrum ordered by inclusion.
+-/
+noncomputable def ringKrullDim (R : Type*) [CommRing R] : WithBot (WithTop ℕ) :=
+  krullDim (PrimeSpectrum R)
+
+variable {R S : Type*} [CommRing R] [CommRing S]
+
+@[nontriviality]
+lemma ringKrullDim_eq_bot_of_subsingleton [Subsingleton R] :
+    ringKrullDim R = ⊥ :=
+  krullDim_eq_bot_of_isEmpty
+
+lemma ringKrullDim_nonneg_of_nontrivial [Nontrivial R] :
+    0 ≤ ringKrullDim R :=
+  krullDim_nonneg_of_nonempty
+
+theorem ringKrullDim_eq_topologicalKrullDim :
+    ringKrullDim R = topologicalKrullDim (PrimeSpectrum R) :=
+  Eq.symm <| krullDim_orderDual.symm.trans <| krullDim_eq_of_orderIso
+  (PrimeSpectrum.pointsEquivIrreducibleCloseds R).symm
+
+/-- If `f : R →+* S` is surjective, then `ringKrullDim S ≤ ringKrullDim R`. -/
+theorem ringKrullDim_le_of_surjective (f : R →+* S) (hf : Function.Surjective f) :
+    ringKrullDim S ≤ ringKrullDim R :=
+  krullDim_le_of_strictMono (PrimeSpectrum.comap f) (Monotone.strictMono_of_injective
+    (fun _ _ hab ↦ Ideal.comap_mono hab) (PrimeSpectrum.comap_injective_of_surjective f hf))
+
+/-- If `I` is an ideal of `R`, then `ringKrullDim (R ⧸ I) ≤ ringKrullDim R`. -/
+theorem ringKrullDim_quotient_le (I : Ideal R) :
+    ringKrullDim (R ⧸ I) ≤ ringKrullDim R :=
+  ringKrullDim_le_of_surjective _ Ideal.Quotient.mk_surjective
+
+/-- If `R` and `S` are isomorphic, then `ringKrullDim R = ringKrullDim S`. -/
+theorem ringKrullDim_eq_of_ringEquiv (e : R ≃+* S) :
+    ringKrullDim R = ringKrullDim S :=
+  le_antisymm (ringKrullDim_le_of_surjective e.symm e.symm.surjective)
+    (ringKrullDim_le_of_surjective e e.surjective)
+
+alias RingEquiv.ringKrullDim := ringKrullDim_eq_of_ringEquiv
+
+section DimensionZero
+
+theorem Ideal.IsPrime.isMaximal_of_ringKrullDim_eq_zero
+    {I : Ideal R} (hI : I.IsPrime) (hdim : ringKrullDim R = 0) :
+    I.IsMaximal :=
+  (PrimeSpectrum.isMaximal_iff ⟨I, _⟩).2 <| mem_maximals_of_krullDim_eq_zero hdim ⟨I, hI⟩
+
+theorem ringKrullDim_eq_zero_iff_forall_isMaximal [Nontrivial R] :
+    ringKrullDim R = 0 ↔ ∀ (I : Ideal R), I.IsPrime → I.IsMaximal := by
+  refine krullDim_eq_zero_iff_forall_mem_maximals_of_nonempty.trans ?_
+  simp_rw [← PrimeSpectrum.isMaximal_iff]
+  exact ⟨(· ⟨·, ·⟩), (· _ <| PrimeSpectrum.isPrime ·)⟩
+
+theorem Ideal.IsPrime.mem_minimalPrimes_of_ringKrullDim_eq_zero
+    {I : Ideal R} (hI : I.IsPrime) (hdim : ringKrullDim R = 0) :
+    I ∈ minimalPrimes R :=
+  (PrimeSpectrum.mem_minimalPrimes_iff ⟨I, _⟩).2 <| mem_minimals_of_krullDim_eq_zero hdim ⟨I, hI⟩
+
+theorem ringKrullDim_eq_zero_iff_forall_mem_minimalPrimes [Nontrivial R] :
+    ringKrullDim R = 0 ↔ ∀ (I : Ideal R), I.IsPrime → I ∈ minimalPrimes R := by
+  refine krullDim_eq_zero_iff_forall_mem_minimals_of_nonempty.trans ?_
+  simp_rw [← PrimeSpectrum.mem_minimalPrimes_iff]
+  exact ⟨(· ⟨·, ·⟩), (· _ <| PrimeSpectrum.isPrime ·)⟩
+
+end DimensionZero
+
+section Field
```

Comment:
Let's split this section into a separate file.

Reply:
Shall I create a folder `Mathlib.RingTheory.KrullDimension` and create the files `Mathlib.RingTheory.KrullDimension.Basic` and `Mathlib.RingTheory.KrullDimension.Field` and then move the content of this section into `Mathlib.RingTheory.KrullDimension.Field`?